src/catala/surface/desugaring.ml

```diff
@@ -728,9 +728,28 @@ and disambiguate_match_and_build_expression (scope : Scopelang.Ast.ScopeName.t)
     (ctxt : Name_resolution.context) (cases : Ast.match_case Pos.marked list) :
     Scopelang.Ast.expr Pos.marked Bindlib.box Scopelang.Ast.EnumConstructorMap.t
     * Scopelang.Ast.EnumName.t =
-  let expr, e_name =
-    List.fold_left
-      (fun (cases_d, e_uid) (case, _pos_case) ->
+  let create_var = function
+    | None -> (ctxt, (Scopelang.Ast.Var.make ("_", Pos.no_pos), Pos.no_pos))
+    | Some param ->
+        let ctxt, param_var = Name_resolution.add_def_local_var ctxt param in
+        (ctxt, (param_var, Pos.get_position param))
+  in
+  let bind_case_body c_uid e_uid ctxt param_pos case_body e_binder =
```

Comment:
I would prefer for this kind of functions that the type of the parameters be annotated, as it really improves the readability of the code.